screens/EpisodeInputScreen.tsx

```diff
@@ -0,0 +1,398 @@
+  EXACT_DATE.getDate(),
+  0
+).toISOString();
+
+const convertToMinutes = (time: Date) => {
+  return time.getHours() * 60 + time.getMinutes();
+};
+
+interface TimeProps {
+  time: Date | undefined;
+  setTime: (time: Date | undefined) => void;
+  setVisibility: (visibility: boolean) => void;
+  isVisible: boolean;
+  label: string;
+  style: StyleProp<ViewStyle>;
+}
+
+function TimePicker({
+  time,
+  setTime,
+  isVisible,
+  setVisibility,
+  label,
+  style,
+}: TimeProps) {
+  const handleConfirm = (
+    date: Date | undefined,
+    setTime: (time: Date | undefined) => void
+  ) => {
+    setTime(date);
+    hideDatePicker();
+  };
+
+  const showDatePicker = () => {
+    setVisibility(true);
+  };
+
+  const hideDatePicker = () => {
+    setVisibility(false);
+  };
+  return (
+    <View style={style}>
+      <Button onPress={showDatePicker} style={datePickerStyles.timeButton}>
+        <View style={datePickerStyles.contentWrapper}>
+          <Text style={datePickerStyles.text}>
+            {time
+              ? time.toLocaleTimeString("en-US", {
+                  hour: "numeric",
+                  minute: "2-digit",
+                })
+              : label}
+          </Text>
+          <Icon
+            name="chevron-down"
+            size={20}
+            color="#000"
+            style={datePickerStyles.icon}
+          />
+        </View>
+      </Button>
+      <DateTimePickerModal
+        headerTextIOS={`Please enter the episode ${label.toLowerCase()}:`}
+        isVisible={isVisible}
+        mode="time"
+        onConfirm={(date) => handleConfirm(date, setTime)}
+        onCancel={hideDatePicker}
+      />
+    </View>
+  );
+}
+
+const datePickerStyles = StyleSheet.create({
+  timeButton: {
+    flex: 1,
+    backgroundColor: "transparent",
+  },
+  contentWrapper: {
+    flex: 1,
+    marginHorizontal: 15,
+    flexDirection: "row",
+    alignItems: "center",
+  },
+  icon: {
+    marginLeft: "auto",
+  },
+  text: {
+    fontFamily: "Arimo",
+    fontStyle: "normal",
+    fontWeight: "normal",
+    fontSize: 18,
+    lineHeight: 22,
+  },
+});
+
+export default function EpisodeInputScreen() {
+  let navigation = useNavigation();
+
+  const [episodeName, setEpisodeName] = useState("");
+  const [initials, setInitials] = useState("");
+  const [startTime, setStartTime] = useState<Date | undefined>(undefined);
+  const [endTime, setEndTime] = useState<Date | undefined>(undefined);
+
+  const [isStartDatePickerVisible, setStartDatePickerVisible] = useState(false);
+  const [isEndDatePickerVisible, setEndDatePickerVisible] = useState(false);
+
+  const [episodes, setEpisodes] = useState<Episode[]>([]);
+
+  let recordingDay: number = 1;
+
+  const addEpisodeToStoage = async (newEpisode: Episode) => {
+    const episodes = await AsyncStorage.getItem(
+      STORAGE_KEYS.daysEpisodes(recordingDay)
+    );
+    if (episodes) {
+      const episodeArray = JSON.parse(episodes);
+      episodeArray.push(newEpisode);
```

Comment:
If we have the episodes state, why do we get the episodes from asyncstorage every time? Or do we even use the episodes state?